main.cpp

```diff
@@ -126,384 +167,529 @@ struct CarWash
 
-    // Increase control of recoil in gun
-    int increaseControlRecoil(int recoilForce, float gunWeight); // Returns percentage of recoil control increase
-    // Toggle folded/unfolded state
-    void toggleFoldedUnfolded(bool isFoldable); //
-    // Provide support to firer's head when ADS
+    float increaseControlRecoil(int recoilForce, float gunWeight); 
+    void toggleFoldedUnfolded(bool isFoldable);
     void supportHeadWhenADS(bool isAiming = true, float pressureAgainstStock = 0.0f);
 };
-/*
-Thing 10) Assault rifle
-5 properties:
-    1) Barrel
-    2) Scope
-    3) Magazine
-    4) Bullet
-    5) Stock
-3 things it can do: 
-    1) Discharge bullet
-    2) Chamber bullet
-    3) Eject empty magazine
- */
+
+float Stock::increaseControlRecoil(int recoilForce, float gunWeight)
+{
+    float controlScore = recoilForce + gunWeight;
+    return controlScore;
+}
+
+void Stock::toggleFoldedUnfolded(bool isStockFoldable)
+{
+    bool toggleCommand = true;
+    if (isStockFoldable == true)
+    {
+        if (toggleCommand == true)
+        {
+            std::cout << "Stock folded\n";
+        }
+        else
+        {
+            std::cout << "Fold command cancel\n";
+        }
+    }
+    else
+    {
+        std::cout << "Unable to fold";
+    }
+    return;
+}
+
+void Stock::supportHeadWhenADS(bool isAiming, float pressureAgainstStock)
+{
+    float supportScore;
+    if (isAiming == true)
+    {
+        supportScore = pressureAgainstStock * pressureAgainstStock;
+    }
+    else
+    {
+        supportScore = 0.0f;
+    }
+    return;
+}
+
 struct AssaultRifle
 { 
-    // Barrel
     Barrel standardAssault;
 
-    // Scope
     Scope standardSAR;
 
-    // Magazine
     Magazine stanagStandard;
 
-    // Bullet
     Bullet standard556;
 
-    // Stock
     Stock bullpup;
  
-    // Discharge bullet
     void dischargeBullet(bool bulletInChamber, bool isJammed);
-    // Chamber bullet
     void chamberBullet(bool magazineLoaded, std::string bulletType = "FMJ");
-    // Eject empty magazine
     void ejectEmptyMagazine(bool isEjectButtonPressed, bool isMagazineEmpty);
 };
 
+void AssaultRifle::dischargeBullet(bool bulletInChamber, bool isJammed)
+{
+    bool dischargeBullet;
+    if (bulletInChamber == true && isJammed == false)
+    {
+        dischargeBullet = true;
+        std::cout << "Bullet discharged";
+    }
+    else
+    {
+        dischargeBullet = false;
+    }
+    std::cout << dischargeBullet;
+    return;
+}
+
+void AssaultRifle::chamberBullet(bool magazineLoaded, std::string bulletType)
+{
+    if (magazineLoaded == true)
+    {
+        std::cout << "Magazine loaded with " << bulletType;
+    }
+    else
+    {
+        std::cout << "Magazine not loaded";
+    }
+    return;
+}
+
+void AssaultRifle::ejectEmptyMagazine(bool isEjectButtonPressed, bool isMagazineEmpty)
+{
+    if (isEjectButtonPressed == true && isMagazineEmpty == true)
+    {
+        std::cout << "Ejecting empty magazine";
+    }
+    return;
+}
```

Comment:
Functions that don't return anything don't need a `return;` statement
- [ ] revise